sdk/core/azure-core/src/main/java/com/azure/core/http/HttpHeaders.java

```diff
@@ -216,28 +220,54 @@ private HttpHeaders setInternal(String formattedName, String name, List<String>
      * words, this will create a header for each key in the provided map, replacing or removing an existing one,
      * depending on the value. If the given values list is null, the header with the given name will be removed. If the
      * given name is already a header, it will be removed and replaced with the headers provided.
+     * <p>
+     * Use {@link #setAllHttpHeaders(HttpHeaders)} if you already have an instance of {@link HttpHeaders} as it provides better
+     * performance.
      *
      * @param headers a map containing keys representing header names, and keys representing the associated values.
      * @return The updated HttpHeaders object
+     * @throws NullPointerException If {@code headers} is null.
      */
     public HttpHeaders setAll(Map<String, List<String>> headers) {
```

Comment:
wonder if this one should be deprecated as well?

Reply:
I'm leaving this as not deprecated as `setAllHttpHeaders` isn't a direct replacement for this. This API is useful for `HttpClient` implementations that track HTTP headers in a Map.